engine/src/test/java/io/zeebe/engine/processor/StreamProcessorHealthTest.java

```diff
@@ -0,0 +1,271 @@
+            processingContext -> {
+              final ZeebeState zeebeState = processingContext.getZeebeState();
+              mockedLogStreamWriter =
+                  new WrappedStreamWriter(processingContext.getLogStreamWriter());
+              processingContext.zeebeState(zeebeState);
+              processingContext.logStreamWriter(mockedLogStreamWriter);
+              return processors(zeebeState.getKeyGenerator())
+                  .onEvent(
+                      ValueType.WORKFLOW_INSTANCE,
+                      ELEMENT_ACTIVATING,
+                      new TypedRecordProcessor<>() {
+                        @Override
+                        public void processRecord(
+                            final long position,
+                            final TypedRecord<UnifiedRecordValue> record,
+                            final TypedResponseWriter responseWriter,
+                            final TypedStreamWriter streamWriter,
+                            final Consumer<SideEffectProducer> sideEffect) {
+
+                          invocation.getAndIncrement();
+                          if (shouldProcessingThrowException.get()) {
+                            throw new RuntimeException("processing failed");
+                          }
+                        }
+                      });
+            });
+
+    return streamProcessor;
+  }
+
+  private final class WrappedStreamWriter implements TypedStreamWriter {
+
+    private final TypedStreamWriter wrappedWriter;
+
+    private WrappedStreamWriter(final TypedStreamWriter wrappedWriter) {
+      this.wrappedWriter = wrappedWriter;
+    }
+
+    @Override
+    public void appendRejection(
+        final TypedRecord<? extends UnpackedObject> command,
+        final RejectionType type,
+        final String reason) {
+      wrappedWriter.appendRejection(command, type, reason);
+    }
+
+    @Override
+    public void appendRejection(
+        final TypedRecord<? extends UnpackedObject> command,
+        final RejectionType type,
+        final String reason,
+        final Consumer<RecordMetadata> metadata) {
+      wrappedWriter.appendRejection(command, type, reason, metadata);
+    }
+
+    @Override
+    public void appendNewEvent(final long key, final Intent intent, final UnpackedObject value) {
+      wrappedWriter.appendNewEvent(key, intent, value);
+    }
+
+    @Override
+    public void appendFollowUpEvent(
+        final long key, final Intent intent, final UnpackedObject value) {
+      wrappedWriter.appendFollowUpEvent(key, intent, value);
+      if (shouldFailErrorHandlingInTransaction.get()) {
+        throw new RuntimeException("append followup event failed");
+      }
+    }
+
+    @Override
+    public void appendFollowUpEvent(
+        final long key,
+        final Intent intent,
+        final UnpackedObject value,
+        final Consumer<RecordMetadata> metadata) {
+      if (shouldFailErrorHandlingInTransaction.get()) {
```

Comment:
Is there a reason why in this method it is done before calling the write in the method above after calling?

Reply:
No. Just copy paste error :grin: It should throw exception with out invoking the actual method.